weave-js/src/components/Panel2/ChildPanelExportReport/utils.ts

```diff
@@ -0,0 +1,70 @@
+import {useBranchPointFromURIString} from '../../PagePanelComponents/hooks';
+import {uriFromNode, determineURISource} from '../../PagePanelComponents/util';
+import {ChildPanelFullConfig} from '../ChildPanel';
+import {format, getYear} from 'date-fns';
+
+export const CREATE_NEW_REPORT_OPTION = 'Create new report';
+export const DEFAULT_REPORT_OPTION = {
+  id: null,
+  name: CREATE_NEW_REPORT_OPTION,
+  projectName: null,
+};
+
+export type EntityOption = {
+  name: string;
+  isTeam: boolean;
+};
+export type ReportOption = {
+  id?: string;
+  name: string;
+  projectName?: string;
+  updatedAt?: number;
+};
+
+export function useEntityAndProject(config: ChildPanelFullConfig) {
+  const inputNode = config.input_node;
+  const maybeURI = uriFromNode(inputNode);
+  const branchPoint = useBranchPointFromURIString(maybeURI);
+  const entityProjectName = determineURISource(maybeURI, branchPoint);
+
+  return {
+    entityName: entityProjectName?.entity ?? '',
+    projectName: entityProjectName?.project ?? '',
+  };
+}
```

Comment:
I also included projectName in here because we'll be needing it to create the core app report url. However; there seems to be an `opReportLink` that I want to explore as well to see what that op does exactly. It might do what we want